lighthouse-core/audits/error-logs.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Audits a page to determine if it is calling deprecated APIs.
+ * This is done by collecting console log messages and filtering them by ones
+ * that contain deprecated API warnings sent by Chrome.
+ */
+
+const Audit = require('./audit');
+
+class ErrorLogs extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'ErrorLogs',
+      name: 'error-logs',
+      description: 'No errors logged to the console',
+      helpText: `
+        Errors logged to the console indicate unresolved problems.
+        They can come from JavaScript exceptions, network request failures,
+        or other browser concerns.
+      `,
+      failureDescription: 'Errors were logged to the console',
+      requiredArtifacts: ['ChromeConsoleMessages']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const entries = artifacts.ChromeConsoleMessages;
+    const tableRows = entries.filter(log => log.entry.level === 'error').map(item => {
```

Comment:
`artifacts.ChromeConsoleMessages` actually excludes javascript exceptions. You might have noticed this when you were testing. I didn't realize it until I tested this today. :/ whoops.

The exceptions will come via [`Runtime.exceptionThrown`](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#event-exceptionThrown`). So we'll need another gatherer to retrieve these, then we can merge the ChromeConsoleMessages together with them.

This will certainly increase the scope of this, so **it probably makes sense to split it into a separate PR?**
If you're OK with that... i'll suggest a few changes for this so we can get it landed, and then work on the exceptions.

sgty?

Reply:
good spot.. and yes I didn't notice it :-(
definitely this has to be done to avoid providing only half of the information but I agree in a separate PR to avoid making the scope of this too big

Reply:
@paulirish I've actioned all your comments, hope we can merge it now